### PR TITLE
Fix issues on 32bit

### PR DESCRIPTION
**Pull request by andreasnoack**
Continuation of #120

### CODE REVIEW

**Comment by andreasnoack** [comment]
Any idea what is going on with the preconditioner tests here. Why are they so much off on 32 bit?

**Comment by xukai92** [comment]
Thanks for the PR. I suspect the failed test might be related to the term we added to stablise the online estimation at https://github.com/TuringLang/AdvancedHMC.jl/blob/master/src/adaptation/precond.jl#L62. I will take a closer look tomorrow.

**Comment by andreasnoack** [comment]
Indeed there was integer overflow in that function on 32 bit. Test are now passing on 32 bit.

File: src/adaptation/precond.jl [hunk 1]
```diff
@@ -57,9 +57,9 @@ end
 
 # https://github.com/stan-dev/stan/blob/develop/src/stan/mcmc/var_adaptation.hpp
 function get_var(wv::WelfordVar{T, AT}) where {T<:AbstractFloat, AT<:AbstractVector{T}}
-    n, M = wv.n, wv.M
+    n, M = T(wv.n), wv.M
```
Comment (xukai92) [inline src/adaptation/precond.jl]: Thanks for the fix. Looks correct to me.

**Comment by xukai92** [comment]
Will merge it in 12 hours.

**Pull request merged**